### PR TITLE
GitHub App configuration support

### LINKED ISSUE #11: Support Github App Values
Hello,

We would like to use this chart with our Github authentication being a Github App instead of a common user.

Would you like a PR for this?

### PR DESCRIPTION
**Pull request by blacksd**
Hi all,
this PR adds the support for GitHub App in the configuration; it will resolve #11. 
The chart must be deployed initially with a set of fake values to be able to invoke the `/github-app/setup`; when the App itself has been created, one can upgrade the installation with the correct values.
Please review and test as needed, I'll be happy to continue the work on this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ This chart creates a single pod in a StatefulSet running Atlantis.  Atlantis per
 In order for Atlantis to start and run successfully:
 1. At least one of the following sets of credentials must be defined:
     - `github`
+    - `github_app`
```
Comment (lkysow) [inline README.md:16]: use camelcase, e.g. `githubApp`

File: charts/atlantis/values.yaml [hunk 2]
```diff
@@ -19,6 +19,20 @@ orgWhitelist: <replace-me>
 #   hostname: github.your.org
 # (The chart will perform the base64 encoding for you for values that are stored in secrets.)
 
+# If using a GitHub App, please enter your values as follows:
+# github_app:
```
Comment (lkysow) [inline charts/atlantis/values.yaml:23]:
```suggestion
# githubApp:
```

**Comment by m1racoli** [comment]
@blacksd do you think you'll be able to address the PR comments soon? I'd be really happy if this becomes available soon. :)

**Comment by blacksd** [comment]
@m1racoli sure, I'll get to it later today 👍🏻

File: README.md (hunk 1 above)
Comment (blacksd) [reply README.md]: This has been taken care of in edac576.

File: charts/atlantis/values.yaml (hunk 2 above)
Comment (blacksd) [reply charts/atlantis/values.yaml]: This has been taken care of in edac576.

File: README.md [hunk 3]
```diff
@@ -39,77 +40,79 @@ extraManifests:
...
+| `test.enabled`                              | Whether to enable the test.                                                                                                                                                                                                                                                                               | `true`                |
+| `extraManifests`                            | add additional manifests to deploy                                                                                                                                                                                                                                                                        | `[]`                  |
+| `initContainers`                            | Containers used to initialize context for Atlantis pods                                                                                                                                                                                                                                                   | `[]`                  |
```
Comment (lkysow) [inline README.md:112]: hey can you do the formatting in a separate PR?

**Review by lkysow — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (blacksd) [reply README.md]: Sure, I reverted it from `main`.

**Comment by AndreZiviani** [comment]
Is there something pending here? Can I help somehow?

**Review by romulofranca — DISMISSED** [review]

File: README.md (hunk 3 above)
Comment (blacksd) [reply README.md]: Sorry for the delay; I reverted original the formatting and added only the new field description in 56c4b8f

**Comment by blacksd** [comment]
Apologies for the delay; I fixed the `README` and we should be good, now.

**Comment by AndreZiviani** [comment]
sorry to bump this again but can you review @lkysow ? :pray:

**Comment by lkysow** [comment]
Hi, sorry I had to approve the workflow for it to run. Please fix the error.

File: README.md [hunk 4]
```diff
@@ -64,6 +64,9 @@ The following options are supported.  See [values.yaml](values.yaml) for more de
...
+| `githubApp.id`                              | GitHub app ID. If set, GitHub authentication will be performed as an [installation](https://developer.github.com/v3/apps/installations/).                                                                                                                                                                 | n/a     |
+| `githubApp.key`                             | A GitHub App PEM encoded private key file. If set, GitHub authentication will be performed as an [installation]((https://developer.github.com/v3/apps/installations/)).                                                                                                                                   | n/a     |
+| `githubApp.secret`                          | Secret used to validate GitHub webhooks (see https://developer.github.com/webhooks/securing/).                                                                                                                                                                                                            | n/a     |
```
Comment (AndreZiviani) [inline README.md:69]:
```suggestion
| `githubApp.secret`                          | Secret used to validate GitHub webhooks (see [securing](https://developer.github.com/webhooks/securing/)).                                                                                                                                                                                                            | n/a     |
```

**Review by AndreZiviani — CHANGES_REQUESTED** [review]
@blacksd :pray:
